source/features/useful-not-found-page.tsx

```diff
@@ -28,119 +25,360 @@ async function checkAnchor(anchor: HTMLAnchorElement): Promise<void> {
 		} else {
-			const pathname = '/' + parts.slice(0, i + 1).join('/');
-			bar.append(i ? ' / ' : '', <a href={pathname}>{part}</a>);
+			// Else just show a hyperlink to that part of the link
+			const pathname = "/" + parts.slice(0, i + 1).join("/");
+			bar.append(i ? " / " : "", <a href={pathname}>{part}</a>);
 		}
 	}
 
-	select('main > :first-child, #parallax_illustration')!.after(bar);
+	select("main > :first-child, #parallax_illustration")!.after(bar);
 
 	// Check parts from right to left; skip the last part
 	for (let i = bar.children.length - 2; i >= 0; i--) {
+		// Strikethrough the parts that don't exist
 		void checkAnchor(bar.children[i] as HTMLAnchorElement);
 	}
+}
 
-	if (['tree', 'blob'].includes(parts[2])) {
-		// Object might be 410 Gone
-		void addCommitHistoryLink(bar);
-	}
+function init(): false | void {
+	// Get the current URL
+	const parts = parseCurrentURL();
 
-	if (['tree', 'blob', 'edit'].includes(parts[2])) {
-		// File might not be available on the current branch
-		// GitHub already redirects /tree/ and /blob/ natively
-		void addDefaultBranchLink(bar);
+	// Check that this is actually a 404 page for a file/folder in a repo
+	if (
+		parts.length <= 1 ||
+		!select.exists('[alt*="This is not the web page you are looking for"]')
+	) {
+		return false;
 	}
+
+	// Create a bar to show the path of the file and which parts of it caused the 404
+	const bar = <h2 className="container mt-4 text-center" />;
+	void showWhatIsMissingInPath(bar);
+
+	// What happened to the file?
+	// Check if it:
+	// - exists on the default branch
+	// - exists in commit history for the current branch (got deleted/moved/renamed)
+	// - exists in commit history for the default branch (if the above 2 fail)
+	// If not, the file does not exist (wrong URL), we don't need to show any extra info
+	void showAdditionalInformation(bar);
 }
 
 async function initPRCommit(): Promise<void | false> {
-	const commitUrl = location.href.replace(/pull\/\d+\/commits/, 'commit');
+	const commitUrl = location.href.replace(/pull\/\d+\/commits/, "commit");
 	if (await is404(commitUrl)) {
 		return false;
 	}
 
-	const blankSlateParagraph = await elementReady('.blankslate p', {waitForChildren: false});
+	const blankSlateParagraph = await elementReady(".blankslate p", {
+		waitForChildren: false,
+	});
 	blankSlateParagraph!.after(
-		<p>You can also try to <a href={commitUrl}>view the detached standalone commit</a>.</p>,
+		<p>
+			You can also try to{" "}
+			<a href={commitUrl}>view the detached standalone commit</a>.
+		</p>
 	);
 }
 
-void features.add(__filebasename, {
-	include: [
-		pageDetect.is404,
-	],
-	init: onetime(init),
-}, {
-	include: [
-		pageDetect.isPRCommit404,
-	],
-	awaitDomReady: false,
-	init: onetime(initPRCommit),
-});
+void features.add(
+	__filebasename,
+	{
+		include: [pageDetect.is404],
+		init: onetime(init),
+	},
+	{
+		include: [pageDetect.isPRCommit404],
+		awaitDomReady: false,
+		init: onetime(initPRCommit),
+	}
+);
```

Comment:
Please don't run Prettier on the code, use the included linter. You can run `npm run fix` and undo unrelated changes like this one

Reply:
Oh sorry about that, I use the prettier extension in VSCode. I'll revert it right away.

Reply:
This chunk, like others, is still affected by the PR. Almost every comment has been moved up by a line